model/task_lifecycle.go

```diff
@@ -31,109 +31,114 @@ type StatusChanges struct {
+				grip.Critical(message.Fields{
+					"message": "task is missing distro id",
+					"task_id": t.Id,
+				})
+			}
 
-		// If the task was not activated by step back, and either the caller is not evergreen
-		// or the task was originally activated by evergreen, deactivate the task
-	} else if !evergreen.IsSystemActivator(caller) || evergreen.IsSystemActivator(t.ActivatedBy) {
-		var err error
-		// deactivate later tasks in the group as well, since they won't succeed without this one
-		if t.IsPartOfSingleHostTaskGroup() {
-			tasksInGroup, err := task.FindTaskGroupFromBuild(t.BuildId, t.TaskGroup)
+			version, err := VersionFindOneId(t.Version)
 			if err != nil {
-				return errors.Wrapf(err, "error finding task group '%s'", t.TaskGroup)
+				return errors.Wrapf(err, "error find associated version")
+			}
+			if version == nil {
+				return errors.Errorf("could not find associated version : `%s`", t.Version)
 			}
-			for _, taskInGroup := range tasksInGroup {
-				if taskInGroup.TaskGroupOrder > t.TaskGroupOrder {
-					originalTasks = append(originalTasks, taskInGroup)
+			versionsToActivate = append(versionsToActivate, *version)
+
+			// If the task was not activated by step back, and either the caller is not evergreen
+			// or the task was originally activated by evergreen, deactivate the task
+		} else if !evergreen.IsSystemActivator(caller) || evergreen.IsSystemActivator(t.ActivatedBy) {
+			var err error
+			// deactivate later tasks in the group as well, since they won't succeed without this one
+			if t.IsPartOfSingleHostTaskGroup() {
+				tasksInGroup, err := task.FindTaskGroupFromBuild(t.BuildId, t.TaskGroup)
+				if err != nil {
+					return errors.Wrapf(err, "error finding task group '%s'", t.TaskGroup)
+				}
+				for _, taskInGroup := range tasksInGroup {
+					if taskInGroup.TaskGroupOrder > t.TaskGroupOrder {
+						originalTasks = append(originalTasks, taskInGroup)
+					}
 				}
 			}
+			if t.Requester == evergreen.MergeTestRequester {
+				if err = DequeueAndRestartForTask(nil, t, message.GithubStateError, caller, fmt.Sprintf("deactivated by '%s'", caller)); err != nil {
+					return err
+				}
+			}
+			tasksToActivate = append(tasksToActivate, originalTasks...)
+		} else {
+			return nil
 		}
-		err = task.DeactivateTasks(originalTasks, caller)
-		if err != nil {
-			return errors.Wrap(err, "error deactivating task")
-		}
-
-		if t.Requester == evergreen.MergeTestRequester {
-			if err = DequeueAndRestartForTask(nil, t, message.GithubStateError, caller, fmt.Sprintf("deactivated by '%s'", caller)); err != nil {
-				return err
+		if t.IsPartOfDisplay() {
+			if err := UpdateDisplayTaskForTask(t); err != nil {
+				return errors.Wrap(err, "problem updating display task")
 			}
 		}
-	} else {
-		return nil
-	}
 
-	if t.IsPartOfDisplay() {
-		if err := UpdateDisplayTaskForTask(t); err != nil {
-			return errors.Wrap(err, "problem updating display task")
+		if err := UpdateBuildAndVersionStatusForTask(t); err != nil {
```

Comment:
I think it doesn't work to call this before actually activating/deactivating the tasks. But also we really only need to call this once per build so we could keep a list of builds that tasksToActivate belongs to, and then iterate through that after the final if/else.